### PR TITLE
clean normalization_inner_outer

### PR DESCRIPTION
**Pull request by liqiangxl**
similar to #923
(1) Transformed private static functions in the InnerOuterPersistentKernelScheduler class into utility functions within an anonymous namespace.
(2) Removed checks/calculations that are unrelated to inner_outer persistent scheduler.

### CODE REVIEW

**Comment by liqiangxl** [comment]
!build

**Comment by liqiangxl** [comment]
!build

**Comment by liqiangxl** [comment]
!build

**Comment by liqiangxl** [comment]
Checked benchmark cases using `tools/compare_codegen.sh -- build/nvfuser_bench --benchmark_filter=NvFuserScheduler --benchmark_repetitions=1 --benchmark_min_time=0`

```
Previous HEAD position was b3051d8f Fix a memory leak. (#936)
Switched to branch 'llu/clean_inner_outer_norm'
Your branch is up to date with 'origin/llu/clean_inner_outer_norm'.


DIFF RESULT:

No difference found
```

**Comment by liqiangxl** [comment]
!build

**Comment by liqiangxl** [comment]
!build

**Comment by liqiangxl** [comment]
!build

**Comment by liqiangxl** [comment]
!build

File: csrc/scheduler/normalization_inner_outer.cpp [hunk 1]
```diff
@@ -793,1186 +456,220 @@ std::shared_ptr<ReductionParams> innerOuterPersistentHeuristic(
...
+    const int64_t estimated_register_count =
+        persistent_buffer_size / bytes_per_register +
+        scheduler_utils::register_overhead;
+    return std::min(
+        estimated_register_count, scheduler_utils::max_registers_per_thread);
+  };
 
-  return rparams;
-}
+  auto getBlocksPerSM = [&](const int64_t threads_per_sm,
```
Comment (naoyam) [inline csrc/scheduler/normalization_inner_outer.cpp:466]: This part shows as changed. Is there any change here? 

Github's diff indicates lots of changes. Maybe it's just false positives? Could you try to minimize the (false?) changes?

Comment (liqiangxl) [reply csrc/scheduler/normalization_inner_outer.cpp:470]: No change. I tried `hide whitespace` and it removes false changes.
image

**Comment by liqiangxl** [comment]
!build

**Review by naoyam — APPROVED** [review]
LGTM

**Comment by naoyam** [comment]
Do you know why the CI diff check shows some diffs?

**Pull request merged**